packages/legend-query-builder/src/components/explorer/QueryBuilderExplorerPanel.tsx

```diff
@@ -977,7 +1082,8 @@ export const QueryBuilderExplorerPanel = observer(
           {propertySearchPanelState.isSearchPanelOpen && (
             <QueryBuilderPropertySearchPanel
               queryBuilderState={queryBuilderState}
-              triggerElement={searchButtonRef.current}
+              triggerElement={searchInputRef.current}
+              clearSearch={() => propertySearchPanelState.resetSearch()}
```

Comment:
```suggestion
              clearSearch={propertySearchPanelState.resetSearch}
```

Reply:
I tried making that change, but then an error gets thrown because the `this` keyword in the `resetSearch` function gets bound to the `QueryBuilderExplorerPanel` object instead of the `QueryBuilderExplorerState` object. I could fix this by doing
```
clearSearch={propertySearchPanelState.resetSearch.bind(
                propertySearchPanelState,
              )}
```
but that doesn't feel any cleaner than just doing
```
clearSearch={() => propertySearchPanelState.resetSearch()}
```